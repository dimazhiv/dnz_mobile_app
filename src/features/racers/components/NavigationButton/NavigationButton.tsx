import React from 'react';
import { StyleSheet, Text, TouchableOpacity } from 'react-native';

export type NavigationButtonProps = {
  title: string;
  onPress: () => void;
};

function NavigationButton({ onPress, title }: NavigationButtonProps) {
  return (
    <TouchableOpacity onPress={onPress}>
      <Text style={styles.buttonText}>{title}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  buttonText: {
    fontSize: 25,
    color: 'blue',
    paddingHorizontal: 10
  }
});

export default NavigationButton;
