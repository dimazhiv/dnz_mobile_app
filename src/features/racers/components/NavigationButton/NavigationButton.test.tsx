import React from 'react';
import { shallow } from 'enzyme';
import NavigationButton from './NavigationButton';

describe('Snapshot testing NavigationButton component', () => {
  test('correct component view', () => {
    const tree = shallow(<NavigationButton title={'<'} onPress={() => {}} />);
    expect(tree).toMatchSnapshot();
  });
});
