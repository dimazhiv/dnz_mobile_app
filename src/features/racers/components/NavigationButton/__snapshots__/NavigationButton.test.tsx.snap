// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing NavigationButton component correct component view 1`] = `
<TouchableOpacity
  onPress={[Function]}
>
  <Text
    style={
      {
        "color": "blue",
        "fontSize": 25,
        "paddingHorizontal": 10,
      }
    }
  >
    &lt;
  </Text>
</TouchableOpacity>
`;
